apps/emqx/src/emqx_config.erl

```diff
@@ -413,6 +413,31 @@ check_config(SchemaMod, RawConf) ->
     check_config(SchemaMod, RawConf, #{}).
 
 check_config(SchemaMod, RawConf, Opts0) ->
+    try
+        do_check_config(SchemaMod, RawConf, Opts0)
+    catch
+        throw:{Schema, Errors} ->
+            throw(compact_errors(Schema, Errors))
+    end.
+
+%% HOCON tries to be very informative about all the detailed errors
+%% it's maybe too much when reporting to the user
```

Comment:
Would be interesting to have a way to disable this compaction when more info is needed?

Reply:
yeah.
don't feel like it's a high prio at the moment.